pkg/encoding/lorawan/messages.go

```diff
@@ -639,3 +639,30 @@ func GetUplinkMessageIdentifiers(phyPayload []byte) (ttnpb.EndDeviceIdentifiers,
 		return ttnpb.EndDeviceIdentifiers{}, errUnknown("MType")(mhdr.MType.String())
 	}
 }
+
+// GetUplinkMessageIdentifiersFromMessage retrieves the EndDeviceIdentifers from a LoRaWAN message (except DeviceID).
+func GetUplinkMessageIdentifiersFromMessage(msg *ttnpb.Message) (ttnpb.EndDeviceIdentifiers, error) {
+	var ids ttnpb.EndDeviceIdentifiers
+	if msg == nil {
+		return ttnpb.EndDeviceIdentifiers{}, errInvalidPayload.New()
+	}
+	switch msg.MHDR.MType {
+	case ttnpb.MType_CONFIRMED_UP, ttnpb.MType_UNCONFIRMED_UP:
+		payload := msg.GetMACPayload()
+		ids.DevAddr = &types.DevAddr{}
+		(&(payload.DevAddr)).Copy(ids.DevAddr)
```

Comment:
What is this `(&(payload.DevAddr))` for?

Reply:
Copy is only available for (*types.DevAddr). Same below. This is not too clean. Any suggested alternatives? I suppose keeping the old behavior here (retrieve ids from the RawPayload) would be easier and a bit cleaner?

Reply:
To be honest, this solves many troubles with zero vs nil identifiers as well.

Reply:
Probably yes, leave the old behavior as-is

Reply:
Reverted